src/CoreBundle/private/library/classes/TCMSURLHistory.class.php

```diff
@@ -78,26 +85,35 @@ public function getSimilarHistoryElementIndex(array $newElementParameters): ?int
      */
     public function GetURL()
     {
-        if ($this->index > 0) {
-            $url = $this->aHistory[($this->index - 1)]['url'];
+        $historyIndex = $this->getHistoryIndex();
 
-            return $url;
+        if ($historyIndex > 0 && isset($this->aHistory[$historyIndex - 1])) {
+            return $this->aHistory[$historyIndex - 1]['url'];
         } else {
             return false;
         }
     }
 
+    public function getHistoryIndex()
```

Comment:
We don't return an index but the amount of history items - how about `getHistoryCount`?

There are also multiple occasions in this class where the history elements are counted (old code). Could you call this method instead to be consistent?

Reply:
The old property was an index, so it started with zero and count would start with 1. This change would mean, that all external code that worked with the index now counts wrong, so i decided not to work with the count here.

Reply:
OH... just checked the code.. the method really returns the count and not the index. i'm puzzled... will check it and change the method name...
Need more coffee

Reply:
Looks like the "index" was never an index, so i renamed the method.